### ISSUE TITLE
Responding to a DM from JIRA in Mattermost should create a comment in Jira

### ISSUE THREAD

**Issue by jfrerich** [mattermost/mattermost-plugin-jira#489]
#### Summary
The Jira Bot will send DM messages when a user is mentioned, assigned an issue, or a new comment is added to an issue.  

This ticket is to create a method to respond, from within the Jira Bot DM channel, to these messages and have the comment posted in Jira.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-16942

---
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Plugin: Jira" community channel](https://community.mattermost.com/core/channels/jira-plugin).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/) and our [Plugins Guide](https://developers.mattermost.com/extend/plugins/).

**Comment by jfrerich** [comment]
@levb, let's discuss the proposal in this issue so the ticket owner has a complete picture of the proposals.  

![image](https://user-images.githubusercontent.com/7575921/75846747-dc47ce80-5da2-11ea-9264-1000bc909be3.png)
![image](https://user-images.githubusercontent.com/7575921/75846767-eb2e8100-5da2-11ea-9630-766e1a6e110e.png)
![image](https://user-images.githubusercontent.com/7575921/75846796-03060500-5da3-11ea-8d3e-4931b0bdacda.png)

**Comment by promulo** [comment]
Hi @hanzei @jfrerich is this one ready to be worked on? I'm interested on taking it.

**Comment by jfrerich** [comment]
Hi @promulo!  Yes, you are more than welcome to take this ticket.  If you have suggestions or questions, feel free to post them here, or in the Mattermost Plugin: Jira channel https://community.mattermost.com/core/channels/jira-plugin

**Comment by promulo** [comment]
Great, thanks. I'll give it a shot.

**Comment by levb** [comment]
@aaronrothschild @asaadmahmood your early input may be good in this ticket

**Comment by asaadmahmood** [comment]
If the purpose is for users to know that they can Add a comment to the post, and those will be attached to Jira, then I'm thinking it should be quite obvious that the user can do that right off the bat.

So we can post a Hint intermittently.
"Hint: You can add comments to Jira tickets directly from Mattermost by replying to Jira posts."

I think adding an option to the dropdown may not be ideal, especially if we have a lot of actions attributed to different plugins. Perhaps we can consider that later when we introduce sub menus so we can combine multiple Jira action items.

**Comment by promulo** [comment]
Hi, thanks for the insights. Something I'd like to clarify: the use case here is to make replies sent to the Jira bot exclusively in DMs? Would it make sense to just start a thread with the bot instead?

![image](https://user-images.githubusercontent.com/450960/79392848-565b9d80-7f74-11ea-9f32-52a42bb50521.png)

So in the example above the "bar" would be posted in the ticket as a comment.

**Comment by jfrerich** [comment]
I believe the issue with staring a new thread is that you won't know which Jira ticket to apply the message to.  

For instance, the Jira Bot could post 2 comments, each with a different Jira ticket reference.  Without responding to a specific Bot post, how will you determine which of the two tickets to send the new comment?

**Comment by jfrerich** [comment]
I also like @asaadmahmood's suggestion of posting a hint in the Jira bot message!

"Hint: You can add comments to Jira tickets directly from Mattermost by replying to Jira posts."

**Comment by promulo** [comment]
@jfrerich Thanks for answering. So you're saying that when a thread is started it's not possible for the bot to know to which specific message the answer is coming?

About the hint: where exactly would it show up? In the DMs or in the channels where the bot is installed?

**Comment by levb** [comment]
@promulo Just checking in to see if you are still working on this issue?

**Comment by promulo** [comment]
Hi @levb sorry about disappearing. I did start working on it but had to stop due to other commitments. Also, you'll notice that I left a couple of questions on the comment above yours. I could go on and make my own assumptions but it would be nice to get some clarification on those, too.

TL;DR Yes I'd still like to work on it. :)

**Comment by promulo** [comment]
Hi, I'm afraid that I won't be able to commit the necessary time and effort to implement this after all. 😞 Apologies for keeping it blocked for others to work on it, I guess I underestimated my bandwidth in the end. Next time I'll try to keep it realistic. Again, sorry about that.

**Comment by jfrerich** [comment]
@promulo, no apology required!  Your efforts on other tickets are much appreciated and thanks for letting us know your time constraints regarding this ticket! 

I've opened the ticket back up by adding the `Up For Grabs` label

**Comment by wiersgallak** [comment]
Closing issues due to inactivity. This issue can be re-opened with more interest from our community.
